### PR TITLE
[WIP] Add Windows URI handling

### PR DESCRIPTION
**Pull request by hoffmabc**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -154,6 +154,99 @@ if(fs.existsSync(__dirname + path.sep + ".." + path.sep + "OpenBazaar-Server" +
...
 
+if (process.platform === "win32") {
+  initWin32();
+}
+
+function initWin32() {
+  var Registry = require('winreg');
+
+  var iconPath = path.join(process.env.LOCALAPPDATA, "app.ico");
+  registerProtocolHandlerWin32('ob', 'URL:OpenBazaar URL', iconPath, process.execPath);
```
Comment (jjeffryes) [inline main.js:165]: Shouldn't this and setProtocol be at the bottom of the function?

Comment (hoffmabc) [inline main.js:165]: Doesn't matter I don't think. I got this code from webtorrent-desktop as is.

File: main.js [hunk 2]
```diff
@@ -154,6 +154,99 @@ if(fs.existsSync(__dirname + path.sep + ".." + path.sep + "OpenBazaar-Server" +
...
+  function registerProtocolHandlerWin32 (protocol, name, icon, command) {
+    var protocolKey = new Registry({
+      hive: Registry.HKCU, // HKEY_CURRENT_USER
+      key: '\\Software\\Classes\\' + protocol
+    })    
+
+    setProtocol()
+
+    function setProtocol (err) {
+      if (err) log.error(err.message)
```
Comment (jjeffryes) [inline main.js:197]: If there's an error, should it continue with the rest of the function, or do a return?

File: main.js (hunk 1 above)
Comment (jjeffryes) [inline main.js:165]: There's a potential issue with hoisting, I don't think it hurts anything in this case, but it's better to call the function after it's defined.

**Comment by hoffmabc** [comment]
This is not ready

**Comment by jjeffryes** [comment]
Ah, I thought it was good to go.

**Comment by jjeffryes** [comment]
I've removed the changes in Master.

**Pull request merged**